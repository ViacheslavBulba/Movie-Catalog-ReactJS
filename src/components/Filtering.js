import React from 'react';
import '../css/Filtering.css';

export default function Filtering() {
    throw new Error('I crashed!');

    return (
        <div className="filter-container">
            <button className="filter-option">ALL</button>
            <button className="filter-option">DOCUMENTARY</button>
            <button className="filter-option">COMEDY</button>
            <button className="filter-option">HORROR</button>
            <button className="filter-option">CRIME</button>
        </div>
    );
}
