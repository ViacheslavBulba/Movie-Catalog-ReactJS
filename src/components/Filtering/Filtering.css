.filter-option {
    padding: 0;
    margin-right: 35px;
    height: 40px;
    font-family: 'Source Sans Pro';
    font-size: 18px;
    color: #ffffff;
    background-color: #232323;
    text-transform: uppercase;
}

.filter-active {
    border-bottom: 3px solid #fff;
}

.dropdown-label {
    margin-right: 20px;
    color: #555555;
}

@media screen and (max-width: 704px) {
    .filter-container {
        display: none
    }
}