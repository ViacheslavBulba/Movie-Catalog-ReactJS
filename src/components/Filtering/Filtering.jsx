import React from 'react';
import './Filtering.css';
import PropTypes from 'prop-types';
import { setFilterByGenres, thunkedSetMovies } from '../../store/actions';
import store from '../../store/store';
import { connect } from 'react-redux';

function Filtering(props) {
    const onFilterChange = (value) => {
        let filters = props.genresFilter;
        if (value === 'All') {
            filters = [];
        } else {
            if (filters.includes(value)) {
                filters = filters.filter((k) => k !== value);
            } else {
                filters = [...filters, value];
            }
        }
        store.dispatch(setFilterByGenres(filters));
        store.dispatch(thunkedSetMovies());
    };

    return (
        <div className='filter-container'>
            <button
                className={
                    'filter-option ' +
                    (props.genresFilter.length === 0 ? 'filter-active' : '')
                }
                onClick={() => onFilterChange('All')}
            >
                All
            </button>
            <button
                className={
                    'filter-option ' +
                    (props.genresFilter.includes('Drama')
                        ? 'filter-active'
                        : '')
                }
                onClick={() => onFilterChange('Drama')}
            >
                Drama
            </button>
            <button
                className={
                    'filter-option ' +
                    (props.genresFilter.includes('Fantasy')
                        ? 'filter-active'
                        : '')
                }
                onClick={() => onFilterChange('Fantasy')}
            >
                Fantasy
            </button>
            <button
                className={
                    'filter-option ' +
                    (props.genresFilter.includes('Comedy')
                        ? 'filter-active'
                        : '')
                }
                onClick={() => onFilterChange('Comedy')}
            >
                Comedy
            </button>
            <button
                className={
                    'filter-option ' +
                    (props.genresFilter.includes('Action')
                        ? 'filter-active'
                        : '')
                }
                onClick={() => onFilterChange('Action')}
            >
                Action
            </button>
        </div>
    );
}

const mapStateToProps = (state) => {
    return {
        genresFilter: state.filerByGenres,
    };
};

export default connect(mapStateToProps)(Filtering);

Filtering.propTypes = {
    genresFilter: PropTypes.arrayOf(PropTypes.string.isRequired),
};
