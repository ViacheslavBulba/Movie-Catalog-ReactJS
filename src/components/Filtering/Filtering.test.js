import React from 'react';
import renderer from 'react-test-renderer';
import Filtering from './Filtering';

import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

const mockStore = configureStore();

const initialState = {
    pending: false,
    movies: [],
    error: null,
    sortBy: 'release_date',
    sortOrder: 'desc',
    filerByGenres: [],
    search: '',
    searchBy: 'title',
    movieToOverview: null,
};

const store = mockStore(initialState);

test('snapshot test for filtering', () => {
    const component = renderer.create(
        <Provider store={store}>
            <Filtering />
        </Provider>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
