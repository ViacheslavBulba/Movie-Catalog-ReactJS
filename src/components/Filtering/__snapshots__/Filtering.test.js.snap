// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test for filtering 1`] = `
<div
  className="filter-container"
>
  <button
    className="filter-option filter-active"
    onClick={[Function]}
  >
    All
  </button>
  <button
    className="filter-option "
    onClick={[Function]}
  >
    Drama
  </button>
  <button
    className="filter-option "
    onClick={[Function]}
  >
    Fantasy
  </button>
  <button
    className="filter-option "
    onClick={[Function]}
  >
    Comedy
  </button>
  <button
    className="filter-option "
    onClick={[Function]}
  >
    Action
  </button>
</div>
`;
